src/logic/run-substream.ts

```diff
@@ -122,7 +123,60 @@ async function getConfigVars(config: IConfigComponent) {
   }
 }
 
-export async function runSubstream(
+export async function runSetupSubstream(
+  schema: string,
+  components: Pick<AppComponents, "config">,
+  logger: ILoggerComponent.ILogger,
+  options?: {
+    timeout?: number
+  }
+) {
+  const { config } = components
+  const { DB_CONNECTION_STRING } = await getConfigVars(config, schema)
+  const substreamsCommandArguments: string[] = [
+    "setup",
+    DB_CONNECTION_STRING,
+    `schema-${await config.getString("NETWORK")}.sql`,
+  ]
+
+  const { exitPromise, child } = execCommand(
+    logger,
+    SINK_CLI_COMMAND,
+    substreamsCommandArguments,
+    process.env as any,
+    "./"
+  )
+
+  setTimeout(() => {
+    if (exitPromise.isPending) {
+      try {
+        if (!child.killed) {
+          logger.warn("Process did not finish", {
+            pid: child.pid?.toString() || "?",
+            command: SINK_CLI_COMMAND,
+            args: substreamsCommandArguments.join(" "),
+          } as any)
+          exitPromise.reject(new Error("Process did not finish"))
+          if (!child.kill("SIGKILL")) {
+            logger.error("Error trying to kill child process", {
+              pid: child.pid?.toString() || "?",
+              command: SINK_CLI_COMMAND,
+              args: substreamsCommandArguments.join(" "),
+            } as any)
+          }
+        }
+      } catch (err: any) {
+        console.log("err: ", err)
+        logger.error(err)
+      }
+    }
+  }, options?.timeout || 0)
+
+  return await exitPromise
+}
```

Comment:
This section seems to be the same as the one in the `runSubstream` function. What if we abstract it?

Reply:
Good idea. I ended up removing the entire logic since we won't need any `timeout` logic for the process, it's meant to run forever so the timeout won't be necessary.